### PR TITLE
oded/tfl-947

### PR DESCRIPTION
**Pull request by odeds**

### CODE REVIEW

File: dist/main.css [hunk 1]
```diff
@@ -20442,442 +20442,6 @@ $blueBG8: #75cbe8;
```
Comment (assafshp) [inline dist/main.css]: do we need dist in github ?

File: demo/znkExercise/demo.js [hunk 2]
```diff
@@ -4,7 +4,6 @@
     angular.module('demo', [
         'znk.infra-toefl.znkExerciseQuestionsType',
         'ngAnimate',
-        'znk.infra.znkExercise',
```
Comment (assafshp) [inline demo/znkExercise/demo.js]: why did you remove it?

File: src/components/znkExerciseQuestionsType/directives/textareaDrv.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/**
+ * attrs:
+ */
+
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra-toefl.znkExerciseQuestionsType').directive('znkExerciseTextarea',
```
Comment (assafshp) [inline src/components/znkExerciseQuestionsType/directives/textareaDrv.js]: the name of the directive and the file name are too diffrent....please change it to be persistant

File: src/components/znkExerciseQuestionsType/directives/textareaDrv.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                    };
+
+                    scope.$watch(function(){
+                            return ngModelCtrl.$viewValue;
+                        },
+                        function(newValue, oldValue) {
+                            scope.wordsLength = _getWordsNum(newValue || '');
+
+                            if (scope.wordsLength > options.maxWordsLength) {
+                                ngModelCtrl.$viewValue = '';
```
Comment (assafshp) [inline src/components/znkExerciseQuestionsType/directives/textareaDrv.js]: why do you need this line if in the next line you're assisgning the old value

File: src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        function ($timeout /* , MediaSrv , SharedModalsSrv */, QuestionFormatEnum, ENV) {
+            'ngInject';
+            return {
+                templateUrl: 'components/znkExerciseQuestionsType/templates/timedWriteAnswerStateDrv.html',
+                restrict: 'E',
+                require: '^genericStatesQuestion',
+                link: function (scope, element, attrs, genericStatesQuestionCtrl) {
+                    var counterTime = 30 * 60 * 1000;
+
+                    if(ENV.debug){
```
Comment (assafshp) [inline src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js]: I prefer one line....
var counterTime = ENV.debug ?  10 * 1000 : 30 * 60 * 1000;

File: src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                            scope.d.timer.play = false;
+                            //wait for the timer to stop, otherwise an exception will jump
+                            $timeout(function(){
+                                genericStatesQuestionCtrl.goToNextState();
+                            });
+                            return;
+                        }
+
+                        //if(timeLeft <= 3000 && !timerCountDownSound.beenPlayed){
+                        //    timerCountDownSound.beenPlayed = true;
```
Comment (assafshp) [inline src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js]: do we need this comments?

File: src/components/znkExerciseQuestionsType/templates/timedWriteAnswerStateDrv.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<div class="write-time-answer-wrapper" translate-namespace="WRITE_TIME_ANSWER">
+    <div class="response-time-section" ng-if="::d.showBottomContent">
+       <div class="title" translate=".RESPONSE"></div>
+       <svg-icon class="write-clock-icon" name="exercise-type-clock-icon"></svg-icon>
+       <timer ng-model="d.timer.currTime" play="d.timer.play" type="2" config="d.timer.config"></timer>
```
Comment (assafshp) [inline src/components/znkExerciseQuestionsType/templates/timedWriteAnswerStateDrv.html]: what is type 2?

File: demo/znkExercise/demo.js (hunk 2 above)
Comment (odeds) [inline demo/znkExercise/demo.js]: dependency in znkExerciseQuestionsType

File: src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+                        genericStatesQuestionCtrl.setUserAnswer(text);
+                    };
+
+                    scope.d.finish = function(){
+                        genericStatesQuestionCtrl.goToNextState();
+                    };
+
+                    var question = genericStatesQuestionCtrl.getQuestion();
+                    if(question.questionFormatId === QuestionFormatEnum.TEXT_AUDIO.enum){
+                      //  scope.d.openPassageModal = SharedModalsSrv.openPassageModal.bind(SharedModalsSrv,question.groupData.passageAudioTitle,question.groupData.content);
```
Comment (odeds) [inline src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js]: comment for now, will add SharedModalsSrv or something else later

File: src/components/znkExerciseQuestionsType/directives/textareaDrv.js (hunk 4 above)
Comment (odeds) [inline src/components/znkExerciseQuestionsType/directives/textareaDrv.js]: it's not removing existing value unless first assign empty string.

File: src/components/znkExerciseQuestionsType/directives/textareaDrv.js (hunk 3 above)
Comment (odeds) [inline src/components/znkExerciseQuestionsType/directives/textareaDrv.js]: changed file name for znkExerciseTextarea

File: src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+                    var counterTime = 30 * 60 * 1000;
+
+                    if(ENV.debug){
+                        counterTime = 10 * 1000;
+                    }
+
+                    $timeout(function(){
+                        scope.d.showBottomContent = true;
+                        scope.d.timer.play = true;
+                    },5000);
```
Comment (odeds) [inline src/components/znkExerciseQuestionsType/directives/timedWriteAnswerStateDrv.js]: same as in toefl-mobile, 5 sec in user story also. add to config phase?
